server/src/main/java/org/opensearch/indices/recovery/RecoverySettings.java

```diff
@@ -177,7 +177,14 @@ public class RecoverySettings {
     );
 
     // choose 512KB-16B to ensure that the resulting byte[] is not a humongous allocation in G1.
-    public static final ByteSizeValue DEFAULT_CHUNK_SIZE = new ByteSizeValue(512 * 1024 - 16, ByteSizeUnit.BYTES);
+    public static final Setting<ByteSizeValue> INDICES_RECOVERY_CHUNK_SIZE_SETTING = Setting.byteSizeSetting(
+        "indices.recovery.chunk_size",
+        new ByteSizeValue(512 * 1024 - 16, ByteSizeUnit.BYTES),
+        new ByteSizeValue(0, ByteSizeUnit.BYTES),
```

Comment:
min has to be `1 byte` . Though it is only for testing purpose . But users can erroneously set it to this value, which can cause failed recoveries. 

Probably we can have `setChunkSize` to be used only for testing (which can be set to 1 byte) and the min value for this can be `new ByteSizeValue(512 * 1024 - 16, ByteSizeUnit.BYTES)` .